### ISSUE TITLE
new checks: short (-d "") and long (--date="") option formatting 

### ISSUE THREAD

**Issue by Stripy42** [koalaman/shellcheck#2435]
#### For new checks and feature suggestions
- [ x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I tried on my old version, 0.5.0. So tried online also. Both did not pickup the error. 

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
if [[ "${array[17]}" == 1 || $(date -d ="now - 3 years - 1 month" +%s) -lt $(date -d ="01 ${array[2]} ${array[1]}" +%s) ]]; then 
   <some actions here>
fi

```

#### Here's what shellcheck currently says:
<doesn't pick up this one>

#### Here's what I wanted or expected to see:

In myscript.sh line #:
        if [[ "${array[17]}" == 1 || $(date -d "now - 3 years - 1 month" +%s) -lt $(date -d ="01 ${array[2]} ${array[1]}" +%s) ]]; then
                                                                                                                                      ^-- New error number: for single letter flags space and no = required, and long flags (i.e. --date), no space and equals required.

**Comment by xtqqczze** [comment]
Long options are not POSIX compliant.

Also, the `-d` option for the date command is a GNU extension and may not be available on all systems (e.g., BSD or macOS).
